python/ray/ml/train/data_parallel_trainer.py

```diff
@@ -0,0 +1,184 @@
+import logging
+from typing import Dict, Callable, Optional, Union
+
+from ray.ml.trainer import Trainer
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.train.trainer import GenDataset
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.train import BackendConfig
+from ray.util.annotations import DeveloperAPI
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class DataParallelTrainer(Trainer):
+    """A Trainer for data parallel training.
+
+    You should subclass this Trainer if your Trainer follows SPMD (single program,
+    multiple data) programming paradigm- you want multiple processes to run the same
+    function, but on different data.
+
+    This Trainer runs the function ``train_loop_per_worker`` on multiple Ray
+    Actors. The provided datasets are automatically sharded to each Ray actor running
+    the training function.
```

Comment:
Datasets comment can be removed? Since we're only sharding the train dataset, and this is described below.
```suggestion
    Actors. 
```